src/prefect/cli/deploy.py

```diff
@@ -1086,6 +1086,108 @@ def _load_deploy_configs_and_actions(ci=False) -> Tuple[List[Dict], Dict]:
     return deploy_configs, actions
 
 
+def _handle_pick_deploy_without_name(deploy_configs):
+    # Prompt the user to select one or more deployment configurations
+    selectable_deploy_configs = [
+        deploy_config for deploy_config in deploy_configs if deploy_config.get("name")
+    ]
+    if not selectable_deploy_configs:
+        return []
+    selected_deploy_config = prompt_select_from_table(
+        app.console,
+        "Would you like to use an existing deployment configuration?",
+        [
+            {"header": "Name", "key": "name"},
+            {"header": "Entrypoint", "key": "entrypoint"},
+            {"header": "Description", "key": "description"},
+        ],
+        selectable_deploy_configs,
+        opt_out_message="No, configure a new deployment",
+        opt_out_response=None,
+    )
+    return [selected_deploy_config] if selected_deploy_config else []
+
+
+def _log_missing_deployment_names(missing_names, matched_deploy_configs, names):
+    # Log unfound names
+    if missing_names:
+        app.console.print(
+            (
+                "The following deployment(s) could not be found and will not be"
+                f" deployed: {', '.join(list(sorted(missing_names)))}"
+            ),
+            style="yellow",
+        )
+    if not matched_deploy_configs:
+        app.console.print(
+            (
+                "Could not find any deployment configurations with the given"
+                f" name(s): {', '.join(names)}. Your flow will be deployed with a"
+                " new deployment configuration."
+            ),
+            style="yellow",
+        )
+
+
+def _filter_matching_deploy_config(name, deploy_configs):
+    # Logic to find the deploy_config matching the given name
+    # This function handles both "flow-name/deployment-name" and just "deployment-name"
+    matching_deployments = []
+    if "/" in name:
+        flow_name, deployment_name = name.split("/")
+        flow_name = flow_name.replace("-", "_")
+        matching_deployments = [
+            deploy_config
+            for deploy_config in deploy_configs
+            if deploy_config.get("name") == deployment_name
+            and deploy_config.get("entrypoint", "").split(":")[-1] == flow_name
```

Comment:
I think this matching logic will fail if the user gives a custom flow name in the `@flow` decorator.

Reply:
Yeah, it would fail if the user provides a custom flow name. I don't know that we currently store that info even if they save their configuration though? 🤔